src/Operations.jl

```diff
@@ -428,7 +428,7 @@ function deps_graph(env::EnvCache, registries::Vector{Registry.RegistryInstance}
             # unregistered stdlib we must special-case it here.  This is further
             # complicated by the fact that we can ask this question relative to
             # a Julia version.
-            if is_unregistered_stdlib(uuid) || uuid_is_stdlib
+            if (is_unregistered_stdlib(uuid) || uuid_is_stdlib) && haskey(stdlibs(), uuid)
```

Comment:
```suggestion
            if (is_unregistered_stdlib(uuid) || uuid_is_stdlib) && haskey(stdlibs(), uuid)
                # if the dependency is still a stdlib in the current running version of Julia it doesn't need to be downloaded
```

@barche does this seem correct?

Although, I don't understand how the correct version is installed for the target julia version if it's different to the one provided by the host Julia

Reply:
Yes, I think this is a correct summary. It's true that there could be edge cases where this gets an incompatible version of an stdlib, but only in the case where a package is requested for a julia version that is different from the running version. I don't know how to deal with that and I'd propose to tackle that problem when and if it ever occurs.

Reply:
@staticfloat it'd be good to get your take on this part too. 

> Although, I don't understand how the correct version is installed for the target julia version if it's different to the one provided by the host Julia